test/src/test/kotlin/graphql/nadel/tests/hooks/polymorphic-hydration-hook-using-alias-helper.kt

```diff
@@ -57,3 +68,46 @@ class `batch-polymorphic-hydration-actor-fields-are-in-the-same-service` : Polym
 @UseHook
 class `batch-polymorphic-hydration-actor-fields-are-in-the-same-service-return-types-implement-same-interface` :
     PolymorphicHydrationWithAliasTestHook()
+
+@UseHook
+class `new-batch-polymorphic-hydrations-are-executed-in-parallel` : PolymorphicHydrationWithAliasTestHook() {
+    private val coroutineScope = CoroutineScope(Dispatchers.Default)
+
+    override fun makeNadel(builder: Nadel.Builder): Nadel.Builder {
+        val defaultServiceExecutionFactory = builder.serviceExecutionFactory
+        val servicesExecuted = Collections.synchronizedSet(HashSet<String>())
+
+        return super.makeNadel(builder)
+            .serviceExecutionFactory(
+                object : ServiceExecutionFactory {
+                    override fun getServiceExecution(serviceName: String): ServiceExecution {
+                        val defaultServiceExecution = defaultServiceExecutionFactory.getServiceExecution(serviceName)
+
+                        return ServiceExecution { params ->
+                            coroutineScope
+                                .future {
+                                    servicesExecuted.add(serviceName)
+                                    if (params.isHydrationCall) {
+                                        delay(Random.nextInt(400, 800).milliseconds)
+                                        assertTrue(servicesExecuted == hashSetOf("people", "pets", "foo"))
```

Comment:
so by checking that all 3 services have started this wrapped execution code before any of the actual executions started (which happens in the `thenCompose` below), we can assert that the calls happened in parallel?

Reply:
There's random delay but that's just to ensure the service execution order doesn't matter. Either way whenever the first service finishes, the set MUST have all the service names in it.